source/includes/steps-tutorial-dotnet.yaml

```
@@ -0,0 +1,316 @@
+  realm-tutorial-dotnet project. Specifically, you will be working in the following
+  files:
+
+  .. cssclass:: config-table
+  .. list-table::
+     :header-rows: 1
+     :widths: 15 85
+     
+     * - File
+       - Purpose
+
+     * - AddMemberPage.xaml.cs
+       - The screen that enables adding and removing members of a project.
+
+     * - App.xaml.cs
+       - Initializes the app, creates a public static Realm app, and defines 
+         the primary startup page.
+
+     * - EditTaskPage.xaml.cs
+       - Provides the UI for editing an existing Task.
+
+     * - LoginPage.xaml.cs
+       - Provides the functionality to log on with an existing user account, or 
+         to register a new user account. When the app starts, if no user is 
+         currently logged on, this page is displayed first.
+
+     * - ProjectPage.xaml.cs
+       - Provides the UI for viewing all Projects that the user is in.
+
+     * - TaskPage.xaml.cs
+       - Provides the UI for viewing the list of Tasks in a Project..
+
+---
+title: Connect to Your MongoDB Realm App
+ref: connect-to-your-mongodb-realm-app
+content: |
+  To get the app working with your backend, you first need to instantiate the
+  Realm app. The Realm app is the interface to the MongoDB Realm backend.
+  Navigate to the ``App.xaml.cs`` file and complete the ``OnStart()``
+  method:
+
+  .. literalinclude:: /tutorial/generated/code/final/App.codeblock.open-realm.cs
+     :language: csharp
+
+  Be sure to enter your own Realm app ID, which you can :ref:`find in the Realm
+  UI <find-your-app-id>`. 
+
+  At this point, we have a static, public 
+  :dotnet-sdk:`Realms.Sync.App <reference/Realms.Sync.App.html>` object that we 
+  can use elsewhere in the project.
+
+---
+title: Implement the AuthProvider
+ref: implement-the-auth-provider
+content: |
+  Navigate to ``LoginPage.xaml.cs``. In this file, we will implement the logon 
+  functionality for the app. We are using  
+  :dotnet-sdk:`EmailPasswordAuth <reference/Realms.Sync.App.html#Realms_Sync_App_EmailPasswordAuth>` 
+  to authenticate users. To do this, we need to be able to log on existing users 
+  with the 
+  :dotnet-sdk:`LogInAsync() <reference/Realms.Sync.App.html#Realms_Sync_App_LogInAsync_Realms_Sync_Credentials_>`
+  method, and be able to register new users with the 
+  :dotnet-sdk:`RegisterUserAsync() <reference/Realms.Sync.App.EmailPasswordClient.html#Realms_Sync_App_EmailPasswordClient_RegisterUserAsync_System_String_System_String_>`
+  method.
+
+  The ``LoginPage.xaml`` file provides a UI for the user to enter an email address, 
+  password, and then click either the **Log In** or **Create New Account** button. 
+  You will implement the functionality of each button's' ``Clicked`` event.
+
+  In the ``DoLogin()`` method, call ``LogInAsync``, passing the credentials 
+  that the user entered. If that call succeeds, we have a valid 
+  :dotnet-sdk:`Realms.Sync.User <reference/Realms.Sync.User.html`, and we can 
```

```suggestion
  :dotnet-sdk:`Realms.Sync.User <reference/Realms.Sync.User.html>`, and we can 
```

Necessary fix required: broken link